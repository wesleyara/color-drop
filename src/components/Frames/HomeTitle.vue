<template>
  <h1 class="text-5xl">Color Drop</h1>
</template>
