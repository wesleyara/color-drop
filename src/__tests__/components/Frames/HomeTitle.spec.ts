/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, it, expect, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import HomeTitle from "../../../components/Frames/HomeTitle.vue";

describe("HomeTitle unit tests", async () => {
  let sut: VueWrapper<any>;
  const titleMock = "Color Drop";

  beforeEach(async () => {
    sut = mount(HomeTitle);
  });

  it("should render the component", () => {
    const wrapper = sut.exists();

    expect(wrapper).toBeDefined();
  });
  it("should match the snapshot", () => {
    const wrapper = sut.html();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render the component with the correct title", () => {
    const wrapper = sut.text();
    
    expect(wrapper).toContain(titleMock);
  });

  it("should have a class", () => {
    const wrapper = sut.classes();

    expect(wrapper).toContain("text-5xl");
  })
});
