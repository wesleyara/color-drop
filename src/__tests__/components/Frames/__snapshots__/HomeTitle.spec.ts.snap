// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeTitle unit tests > should match the snapshot 1`] = `"<h1 class="text-5xl">Color Drop</h1>"`;
